source/common/http/conn_manager_impl.cc

```diff
@@ -804,16 +805,14 @@ void ConnectionManagerImpl::ActiveStream::decodeData(ActiveStreamDecoderFilter*
 }
 
 HeaderMap& ConnectionManagerImpl::ActiveStream::addDecodedTrailers() {
-  if (state_.filter_call_state_ & FilterCallState::LastDataFrame) {
-    if (request_trailers_) {
-      throw std::logic_error("decodedTrailers added more than once");
-    }
+  // trailers can only be added during the last data frame (i.e. end_stream = true)
+  ASSERT(state_.filter_call_state_ & FilterCallState::LastDataFrame);
 
-    request_trailers_ = std::make_unique<HeaderMapImpl>();
-    return *request_trailers_;
-  } else {
-    throw std::logic_error("addDecodedTrailers called in invalid context");
-  }
+  // traileres can only be added once
```

Comment:
-> Trailers can only be added once.   
here and below.  Also trailers->Trailers above